plugins/woocommerce-admin/client/marketing/overview-multichannel/Campaigns/Campaigns.test.tsx

```diff
@@ -0,0 +1,193 @@
+	it( 'renders a TablePlaceholder when loading is in progress', () => {
+		( useCampaigns as jest.Mock ).mockReturnValue( {
+			loading: true,
+			error: undefined,
+			data: undefined,
+			meta: undefined,
+		} );
+
+		const { container } = render( <Campaigns /> );
+		const tablePlaceholder = container.querySelector(
+			'div.woocommerce-table__table.is-loading'
+		);
+
+		expect( tablePlaceholder ).toBeInTheDocument();
+	} );
+
+	it( 'renders an error message when loading is done and data is undefined', () => {
+		( useCampaigns as jest.Mock ).mockReturnValue( {
+			loading: false,
+			error: {},
+			data: undefined,
+			meta: undefined,
+		} );
+
+		render( <Campaigns /> );
+
+		expect(
+			screen.getByText( 'An unexpected error occurred.' )
+		).toBeInTheDocument();
+	} );
+
+	it( 'renders an info message when loading is done and data is an empty array', () => {
+		( useCampaigns as jest.Mock ).mockReturnValue( {
+			loading: false,
+			error: undefined,
+			data: [],
+			meta: {
+				total: 0,
+			},
+		} );
+
+		render( <Campaigns /> );
+
+		expect(
+			screen.getByText( 'Advertise with marketing campaigns' )
+		).toBeInTheDocument();
+	} );
+
+	it( 'renders a table with campaign info and without pagination when loading is done and data is an array with length no more than page size', () => {
+		( useCampaigns as jest.Mock ).mockReturnValue( {
+			loading: false,
+			error: undefined,
+			data: [
+				{
+					id: `google-listings-and-ads|111`,
+					title: 'Campaign 111',
+					description: '',
+					cost: `USD 111`,
+					manageUrl:
+						'https://wc1.test/wp-admin/admin.php?page=wc-admin&path=/google/dashboard&subpath=/campaigns/edit&programId=111',
+					icon: 'https://woocommerce.com/wp-content/uploads/2021/06/woo-GoogleListingsAds-jworee.png',
+					channelName: 'Google Listings and Ads',
+					channelSlug: 'google-listings-and-ads',
+				},
+			],
+			meta: {
+				total: 1,
+			},
+		} );
+
+		const { container } = render( <Campaigns /> );
+
+		expect( screen.getByText( 'Campaign 111' ) ).toBeInTheDocument();
+
+		expect( screen.getByText( 'USD 111' ) ).toBeInTheDocument();
+
+		const pagination = container.querySelector( '.woocommerce-pagination' );
+		expect( pagination ).not.toBeInTheDocument();
+	} );
+
+	it( 'renders a table with campaign info and with pagination when loading is done and data is an array with length more than page size', async () => {
+		( useCampaigns as jest.Mock ).mockReturnValue( {
+			loading: false,
+			error: undefined,
+			data: [
+				{
+					id: `google-listings-and-ads|111`,
+					title: 'Campaign 111',
+					description: '',
+					cost: `USD 111`,
+					manageUrl:
+						'https://wc1.test/wp-admin/admin.php?page=wc-admin&path=/google/dashboard&subpath=/campaigns/edit&programId=111',
+					icon: 'https://woocommerce.com/wp-content/uploads/2021/06/woo-GoogleListingsAds-jworee.png',
+					channelName: 'Google Listings and Ads',
+					channelSlug: 'google-listings-and-ads',
+				},
```

Comment:
💅  WDYT about a helper `getDummyCampaign( programId )`

Reply:
That would reduce verbosity and repetition

Reply:
```
data: [
  getDummyCampaign( '111' ),
  getDummyCampaign( '222' ),
...
]
```

Reply:
Good idea! Done in [`8e9f344` (#36735)](https://github.com/woocommerce/woocommerce/pull/36735/commits/8e9f344478ebdf50e41bb3efb1f8407580e05ea4).